internal/provider/resource_jaas_role.go

```diff
@@ -0,0 +1,207 @@
+	}
+	resource.client = client
+	// Create the local logging subsystem here, using the TF context when creating it.
+	resource.subCtx = tflog.NewSubsystem(ctx, LogResourceJAASRole)
+}
+
+// Create attempts to create the role represented by the resource in JAAS.
+func (resource *jaasRoleResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	// Check first if the client is configured
+	if resource.client == nil {
+		addClientNotConfiguredError(&resp.Diagnostics, LogResourceJAASRole, "create")
+		return
+	}
+
+	// Read Terraform configuration from the request into the model
+	var plan jaasRoleResourceModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Add the role to JAAS
+	uuid, err := resource.client.Jaas.AddRole(plan.Name.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to add role %q, got error: %s", plan.Name.ValueString(), err))
+		return
+	}
+
+	// Set the UUID in the state
+	plan.UUID = types.StringValue(uuid)
+	resp.Diagnostics.Append(resp.State.Set(ctx, plan)...)
+}
+
+// Read attempts to read the role represented by the resource from JAAS.
+func (resource *jaasRoleResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	// Check first if the client is configured
+	if resource.client == nil {
+		addClientNotConfiguredError(&resp.Diagnostics, LogResourceJAASRole, "read")
+		return
+	}
+
+	// Read the Terraform state from the request into the model
+	var state jaasRoleResourceModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Read the role from JAAS
+	role, err := resource.client.Jaas.ReadRoleByUUID(state.UUID.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to get role %q, got error: %s", state.Name.ValueString(), err))
+		return
+	}
+
+	// Set the role name in the state
+	state.Name = types.StringValue(role.Name)
+	resp.Diagnostics.Append(resp.State.Set(ctx, state)...)
+}
+
+// Update attempts to rename the role represented by the resource in JAAS.
+func (resource *jaasRoleResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	// Check first if the client is configured
+	if resource.client == nil {
+		addClientNotConfiguredError(&resp.Diagnostics, LogResourceJAASRole, "update")
+		return
+	}
+
+	// Read the current state from the request
+	var state jaasRoleResourceModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Read the plan from the request into the model
+	var plan jaasRoleResourceModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// If the name has not changed, there is nothing to do
+	if state.Name.ValueString() == plan.Name.ValueString() {
```

Comment:
This will cover everything include nil. It's the terraform documented approach to comparing for Update.

```suggestion
	if plan.Name.Equal(state.Name) {
```